lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -264,6 +264,7 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     int outputPipe[2]{ -1, -1 };
     if (shouldCaptureOutput) {
       if (::pipe(outputPipe) < 0) {
+        getDelegate().commandProcessStarted(context.job.getForCommand(), handle);
```

Comment:
Yah, this is just to ensure we call `commandProcessStarted` before emitting an error. In the good case, it is being called after `posix_spawn` further down.